cnd/src/swap_protocols/rfc003/bob/mod.rs

```diff
@@ -23,6 +23,7 @@ pub struct State<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset> {
     pub beta_ledger_state: LedgerState<BL, BA>,
     #[derivative(Debug = "ignore")]
     pub secret_source: Arc<dyn DeriveIdentities>,
+    pub failed: bool, // Gets set on any error during the execution of a swap.
```

Comment:
I don't like comments like these.

They scream to being outdated because you describe functionality here that happens somewhere else.
If a developer is interested in when this gets set, they can do a dataflow analysis from this field.

Reply:
Interesting point.  I agree that these comments are susceptible to becoming outdated.  Can I explain how I use these comments in code then perhaps you could tell me if you see holes in the method (or ways to improve).

Case 1: with a comment like this

If I'm interested enough in this area of the codebase, I use the comment as a hint of where to look - then if I find that the comment is correct, win, the comment has helped me quickly find what `failed` field is for.  If I find the comment is incorrect, I will typically have well understood the code by now, enough to be sure I'm wasting peoples time on the mailing list with the patch that I then create removing the comment, win - the code is cleaner and the I understand the field. 

Case 2: with out a comment like this

If I come across a field like this, one that its not immediately obvious when/why its set I have to do, as you suggest, data flow analysis of the containing object.  This is typically a larger scope than in case 1, but on the plus side I guess it means more understanding of the code.  I have to check every where though where as with the hint I can be reasonable confident (as long as the field is not being abused somewhere) where I should look and not bother looking at everysingle call site that modifies the field.

So, my argument is, the comment saves more time than it takes to remove it once it goes stale.  Every developer that hits this declaration is going to wonder what its for so why not give a hint and trust other devs to remove it if it gets stale?

Open to your thoughts

Reply:
Adding a mention to this thread in case you don't see it since its on a merged PR @thomaseizinger.

Reply:
Maybe there is a better name to ths field to?
> Gets set on any error during the execution of a swap.

instead of fail it could be `error_encountered` ?

Reply:
Excuse my brevity as I am on my phone:

- Improved naming might make the comment obsolete
- Your option 1 assumes that I have to by default mistrust every comment, which kind of defeats the purpose IMO.
- Without bashing tools, but getting all call sites with write access to this field are two consecutive shortcuts in CLion 🤷‍♂️. That is probably even faster than reading and understanding the comment.

Inline comments always scream at me: Careful! Non-obvious shit going on here, more attention than usual needed.

Rationale is: if only obvious things were happening, there would be no need to point them out.

The increased understanding of the codebase is a plus aswell :)